ninja-core/src/main/java/ninja/template/TemplateEngineFreemarker.java

```diff
@@ -263,6 +263,8 @@ public void invoke(Context context, Result result) {
         map.put("reverseRoute", templateEngineFreemarkerReverseRouteMethod);
         map.put("assetsAt", templateEngineFreemarkerAssetsAtMethod);
         map.put("webJarsAt", templateEngineFreemarkerWebJarsAtMethod);
+        
+        map.put("authenticity", new TemplateEngineFreemarkerAuthenticityMethod(context.getSession().getAuthenticityToken()));
```

Comment:
Just pass the context. Right now you force the creation of the session cookie and authenticity token for requests that potentially don't use them. The call to getSession().getAuthenticityToken() should be executed only when the template method is called...